### PR TITLE
Refactor model package and update dg

### PR DESCRIPTION
**Pull request by Zhu-Ruicong**
Reorganized the packaging in Model component to make more sense. Developer Guide is updated accordingly. Also included my draft of PPP

### CODE REVIEW

File: docs/team/zhu-ruicong.adoc [hunk 1]
```diff
@@ -0,0 +1,47 @@
+= John Doe - Project Portfolio
```
Comment (shenghaoc) [inline docs/team/zhu-ruicong.adoc:1]: Remember to change your name!

File: docs/DeveloperGuide.adoc [hunk 2]
```diff
@@ -125,6 +125,7 @@ image::DeleteSequenceDiagram.png[]
 
 NOTE: The lifeline for `DeletePetCommandParser` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
 
+// tag::Model[]
```
Comment (shenghaoc) [inline docs/DeveloperGuide.adoc:128]: What do tags do in AsciiDoc? I still don't know

File: docs/diagrams/ModelClassDiagram.puml [hunk 3]
```diff
@@ -68,7 +73,7 @@ Name -[hidden]right-> DateOfBirth
 DateOfBirth -[hidden]right-> Gender
 Species -[hidden]right-> Food
 
-PetTracker *--> "1" Schedule
```
Comment (shenghaoc) [inline docs/diagrams/ModelClassDiagram.puml:71]: Your composition and aggregation seems to be messed up.
I don't think UniquePetList's relationship with Schedule is composition.
And Slot and PetAmountAndPet's relationship with Pet is not composition either.
I'm not sure about the rest but you should check them too.

File: docs/team/zhu-ruicong.adoc (hunk 1 above)
Comment (Zhu-Ruicong) [reply docs/team/zhu-ruicong.adoc:1]: this is still WIP haha

File: docs/DeveloperGuide.adoc (hunk 2 above)
Comment (Zhu-Ruicong) [reply docs/DeveloperGuide.adoc:128]: you can use this tag in another file(in this case in my PPP) and directly call the tag. This entire chunk will be copied over.

File: docs/diagrams/ModelClassDiagram.puml (hunk 3 above)
Comment (Zhu-Ruicong) [reply docs/diagrams/ModelClassDiagram.puml:71]: I just fixed the errors hahaha. Can you glance through again? Thx!

Comment (shenghaoc) [reply docs/diagrams/ModelClassDiagram.puml:71]: PetAmountAndPet looks like an association class to me.
I don't its relationship with pet is "has a".

Comment (Zhu-Ruicong) [reply docs/diagrams/ModelClassDiagram.puml:71]: Hmmmm kinda makes sense. Only doubt I have is that my FoodAmountAndPet(realized there's a type here. gonna change later) doesn't know FoodCollection. Would this violate the point of Association class? The way I see it was more based on the fact that FoodCollection has a whole-part relationship with FoodAmountAndPet, hence the implementation.

Comment (shenghaoc) [reply docs/diagrams/ModelClassDiagram.puml:71]: Why is there no arrow between FoodCollection and Food?
Maybe we can standardize whether to use Integer or int?
FoodAmountAndPet contains Pet? I can't process.

Comment (Zhu-Ruicong) [reply docs/diagrams/ModelClassDiagram.puml:71]: Actually I'm able to remove the association between pet and FoodAmountAndPet. Would this make more sense?

Comment (shenghaoc) [reply docs/diagrams/ModelClassDiagram.puml:71]: Haven't started revising yet so I'm not really sure but intuitively FoodAmountAndPet feels like an association class between FoodAmount and Pet.
https://stackoverflow.com/questions/37767328/implementation-of-classes-with-association-in-java
An association class would be able to have both as variables. However, FoodAmount is just an integer here.

**Comment by shenghaoc** [comment]
Are you sure FoodAmountAndPet cannot exist without FoodCollection?

**Pull request merged**